schedule/schedule-appengine/src/test/java/com/jasify/schedule/appengine/spi/ActivityEndpointTest.java

```diff
@@ -0,0 +1,517 @@
+package com.jasify.schedule.appengine.spi;
+
+import com.google.api.server.spi.auth.common.User;
+import com.google.api.server.spi.response.BadRequestException;
+import com.google.api.server.spi.response.ForbiddenException;
+import com.google.api.server.spi.response.NotFoundException;
+import com.google.api.server.spi.response.UnauthorizedException;
+import com.google.appengine.api.datastore.Key;
+import com.jasify.schedule.appengine.TestHelper;
+import com.jasify.schedule.appengine.model.EntityNotFoundException;
+import com.jasify.schedule.appengine.model.FieldValueException;
+import com.jasify.schedule.appengine.model.UserContext;
+import com.jasify.schedule.appengine.model.activity.*;
+import com.jasify.schedule.appengine.model.common.Organization;
+import com.jasify.schedule.appengine.model.common.OrganizationService;
+import com.jasify.schedule.appengine.model.common.OrganizationServiceFactory;
+import com.jasify.schedule.appengine.model.common.TestOrganizationServiceFactory;
+import com.jasify.schedule.appengine.spi.dm.JasAddActivityTypeRequest;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.easymock.IAnswer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slim3.datastore.Datastore;
+import org.slim3.datastore.EntityNotFoundRuntimeException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jasify.schedule.appengine.spi.JasifyEndpointTest.newCaller;
+import static junit.framework.TestCase.assertEquals;
+import static org.easymock.EasyMock.*;
+
+/**
+ * Created by wszarmach on 19/01/15.
+ */
+public class ActivityEndpointTest {
+
+    private TestActivityServiceFactory testActivityServiceFactory = new TestActivityServiceFactory();
+    private TestOrganizationServiceFactory testOrganizationServiceFactory = new TestOrganizationServiceFactory();
+
+    private ActivityEndpoint endpoint = new ActivityEndpoint();
+
+    @Before
+    public void before() {
+        TestHelper.initializeDatastore();
+        testActivityServiceFactory.setUp();
+    }
+
+    @After
+    public void after() {
+        TestHelper.cleanupDatastore();
+        UserContext.clearContext();
+        testActivityServiceFactory.tearDown();
```

Comment:
OH!  I think this is where you leak was I would strongly recommend calling
```java
testOrganizationServiceFactory.tearDown(false);
```
I see you finally{}'ed it, but still, safer...
Just thought of a potential problem...  if you have to TestXXXFactories and one throws an exception on the tearDown, the other tearDown is never called...
I wonder if this works
```java
@After
public void afterActivity(){
  testactivityService.tearDown();
}
@After
public void afterOrg(){
  testorgService.tearDown();
}
```
But it might be best to change the idea of doing the verify on the tearDown, and have the standard practice
```java
@After
public void after() {
    service1.tearDown(); // never throws
    service2.tearDown(); // never throws
    service1.verfiy();
   service2.verify();
}
```

The oher option would be to have
```
class TestHelper {
  public void tearDown(TestServiceFactory ... factories) {
    foreach(){ 
       teardown;
       verify;
    }
  }
}
```

Well... Still good.